system/lib/libc/musl/src/internal/libc.c

```diff
@@ -7,3 +12,24 @@ char *__progname=0, *__progname_full=0;
 
 weak_alias(__progname, program_invocation_short_name);
 weak_alias(__progname_full, program_invocation_name);
+
+__attribute__((constructor))
+static void __progname_ctor(void)
+{
+	static char full_path[PATH_MAX];
+	char *basename;
+
+	EM_ASM({
+		stringToUTF8(thisProgram, $0, $1);
+	}, full_path, sizeof(full_path));
```

Comment:
Can think its best to avoid EM_ASM/EM_JS in core system libraries for now in favor of adding new library.js function.

How about declaring `_emscipten_get_progname(const char* str, int len)` there and then defining it in `src/library.js`?